discovery/gossiper.go

```diff
@@ -2984,3 +2752,232 @@ func (d *AuthenticatedGossiper) handleNodeAnnouncement(nMsg *networkMsg,
 	// TODO(roasbeef): get rid of the above
 	return announcements, true
 }
+
+// handleChanAnnouncement processes a new channel announcement.
+func (d *AuthenticatedGossiper) handleChanAnnouncement(nMsg *networkMsg,
+	ann *lnwire.ChannelAnnouncement,
+	ops []batch.SchedulerOption) ([]networkMsg, bool) {
+
+	// We'll ignore any channel announcements that target any chain other
+	// than the set of chains we know of.
+	if !bytes.Equal(ann.ChainHash[:], d.cfg.ChainHash[:]) {
+		err := fmt.Errorf("ignoring ChannelAnnouncement from chain=%v"+
+			", gossiper on chain=%v", ann.ChainHash,
+			d.cfg.ChainHash)
+		log.Errorf(err.Error())
+
+		key := newRejectCacheKey(
+			ann.ShortChannelID.ToUint64(),
+			sourceToPub(nMsg.source),
+		)
+		_, _ = d.recentRejects.Put(key, &cachedReject{})
+
+		nMsg.err <- err
+		return nil, false
+	}
+
+	// If the advertised inclusionary block is beyond our knowledge of the
+	// chain tip, then we'll ignore it for now.
+	d.Lock()
+	if nMsg.isRemote && d.isPremature(ann.ShortChannelID, 0, nMsg) {
+		log.Warnf("Announcement for chan_id=(%v), is premature: "+
+			"advertises height %v, only height %v is known",
+			ann.ShortChannelID.ToUint64(),
+			ann.ShortChannelID.BlockHeight, d.bestHeight)
+		d.Unlock()
+		nMsg.err <- nil
+		return nil, false
+	}
+	d.Unlock()
+
+	// At this point, we'll now ask the router if this is a zombie/known
+	// edge. If so we can skip all the processing below.
+	if d.cfg.Router.IsKnownEdge(ann.ShortChannelID) {
+		nMsg.err <- nil
+		return nil, true
+	}
+
+	// If this is a remote channel announcement, then we'll validate all
+	// the signatures within the proof as it should be well formed.
+	var proof *channeldb.ChannelAuthProof
+	if nMsg.isRemote {
+		if err := routing.ValidateChannelAnn(ann); err != nil {
+			err := fmt.Errorf("unable to validate announcement: "+
```

Comment:
nit: fits in 80 characters

Reply:
same here on my IDE it is at 80 characters